src/conn/conn_dhandle.c

```diff
@@ -134,7 +134,17 @@ __conn_dhandle_mark_dead(WT_SESSION_IMPL *session)
 	 * eviction and drain any blocks already queued.
 	 */
 	WT_RET(__wt_evict_file_exclusive_on(session, &evict_reset));
+
+	/* The handle is dead. */
 	F_SET(session->dhandle, WT_DHANDLE_DEAD);
+
+	/*
+	 * Reset the tree's eviction priority, and the tree is evictable by
+	 * definition.
+	 */
+	__wt_evict_priority_clear(session);
+	F_CLR(S2BT(session), WT_BTREE_NO_EVICTION);
```

Comment:
I think there is a bug in ``__wt_evict_file_exclusive_on`` - it should clear the ``WT_BTREE_NO_EVICTION`` flag before returning an error. Otherwise the current behavior is expected - sometimes we want to clear ``WT_BTREE_NO_EVICTION`` so that eviction keeps looking at the file and sometimes we are planning to flush the file ourselves and need eviction to be locked out.